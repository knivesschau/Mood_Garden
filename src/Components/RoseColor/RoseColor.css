/* Mobile styles first, 320px */

/* img:hover {s
    cursor: pointer;
    animation: rotation 1s infinite linear;
}

@keyframes rotation {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(359deg);
    }
} */ 