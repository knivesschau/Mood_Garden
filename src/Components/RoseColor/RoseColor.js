import React, {Component} from 'react';
import moodGardenContext from '../../moodGardenContext';
import roseImg from '../../images/rose-image.png';
import './RoseColor.css';

export default class RoseColor extends Component {
    static contextType = moodGardenContext;

    pickColor(color) {
        switch(color) {
            case "Red": 
                return "hue-rotate(0deg)";
            
            case "Yellow": 
                return "hue-rotate(75deg) saturate(2.50)";
            
            case "White": 
                return "grayscale(100%) brightness(210%)";

            case "Pink": 
                return "hue-rotate(335deg)";

            case "Purple":
                return "hue-rotate(275deg)";

            case "Black":
                return "hue-rotate(0deg) saturate(0%)";
        }
        return "hue-rotate(0deg)"; //default case
    }

    render() {
        const {color} = this.props;

        return (
            <>
                <img 
                    alt="rose-color" 
                    id={color}
                    src={roseImg}
                    style={{width: "20%", filter: this.pickColor(color)}}
                />
            </>
        );
    }
}