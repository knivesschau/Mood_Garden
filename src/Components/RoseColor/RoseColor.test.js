import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RoseColor from './RoseColor';

describe ('RoseColor Component', () => {
    it ('renders the complete RoseColor component image', () => {
        const wrapper = shallow(<RoseColor/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})