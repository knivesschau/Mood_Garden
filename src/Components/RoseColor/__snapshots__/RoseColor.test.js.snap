// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoseColor Component renders the complete RoseColor component image 1`] = `
<Fragment>
  <img
    alt="rose-color"
    src="rose-image.png"
    style={
      Object {
        "filter": "hue-rotate(0deg)",
        "width": "20%",
      }
    }
  />
</Fragment>
`;
